File: 백준/Bronze/25304. 영수증/영수증.py
X = int(input())
N = int(input())
i = 0
a = 0
b = 0
answer = 0
while True:
    if i >= N:
        break
    a, b = map(int, input().split())
    answer += a * b
    i += 1
if answer == X:
    print("Yes")
else:
    print("No")

File: 백준/Bronze/25304. 영수증/영수증.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');
const X = parseInt(input[0]);
const N = parseInt(input[1]);

let answer = 0;

for (let i = 2; i < 2 + N; ++i) {
    const [a, b] = input[i].split(' ').map(Number);
    answer += a * b;
}

console.log(answer === X ? "Yes" : "No");